xh fails to build from source with GCC 15 on amd64

The package xh with version 0.24.1-r2 fails to build from source with GCC 15 on amd64 with the following error:

```2025/05/31 00:59:04 WARN   cargo:warning=  804 |     onig_st_foreach(t, i_free_name_entry, 0);
2025/05/31 00:59:04 WARN   cargo:warning=      |                        ^~~~~~~~~~~~~~~~~
2025/05/31 00:59:04 WARN   cargo:warning=      |                        |
2025/05/31 00:59:04 WARN   cargo:warning=      |                        int (*)(OnigUChar *, NameEntry *, void *) {aka int (*)(unsigned char *, NameEntry *, void *)}
2025/05/31 00:59:04 WARN   cargo:warning=In file included from oniguruma/src/regparse.c:37:
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, NameEntry *, void *)' {aka 'int (*)(unsigned char *, NameEntry *, void *)'}
2025/05/31 00:59:04 WARN   cargo:warning=   55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
2025/05/31 00:59:04 WARN   cargo:warning=      |                               ^~~~~~~~~~~~~~~~
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/st.h:35:18: note: in definition of macro '_'
2025/05/31 00:59:04 WARN   cargo:warning=   35 | # define _(args) args
2025/05/31 00:59:04 WARN   cargo:warning=      |                  ^~~~
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/regparse.c:789:1: note: 'i_free_name_entry' declared here
2025/05/31 00:59:04 WARN   cargo:warning=  789 | i_free_name_entry(UChar* key, NameEntry* e, void* arg ARG_UNUSED)
2025/05/31 00:59:04 WARN   cargo:warning=      | ^~~~~~~~~~~~~~~~~
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/regparse.c: In function 'onig_foreach_name':
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/regparse.c:873:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
2025/05/31 00:59:04 WARN   cargo:warning=  873 |     onig_st_foreach(t, i_names, (HashDataType )&narg);
2025/05/31 00:59:04 WARN   cargo:warning=      |                        ^~~~~~~
2025/05/31 00:59:04 WARN   cargo:warning=      |                        |
2025/05/31 00:59:04 WARN   cargo:warning=      |                        int (*)(OnigUChar *, NameEntry *, INamesArg *) {aka int (*)(unsigned char *, NameEntry *, INamesArg *)}
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, NameEntry *, INamesArg *)' {aka 'int (*)(unsigned char *, NameEntry *, INamesArg *)'}
2025/05/31 00:59:04 WARN   cargo:warning=   55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
2025/05/31 00:59:04 WARN   cargo:warning=      |                               ^~~~~~~~~~~~~~~~
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/st.h:35:18: note: in definition of macro '_'
2025/05/31 00:59:04 WARN   cargo:warning=   35 | # define _(args) args
2025/05/31 00:59:04 WARN   cargo:warning=      |                  ^~~~
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/regparse.c:846:1: note: 'i_names' declared here
2025/05/31 00:59:04 WARN   cargo:warning=  846 | i_names(UChar* key ARG_UNUSED, NameEntry* e, INamesArg* arg)
2025/05/31 00:59:04 WARN   cargo:warning=      | ^~~~~~~
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/regparse.c: In function 'onig_renumber_name_table':
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/regparse.c:901:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
2025/05/31 00:59:04 WARN   cargo:warning=  901 |     onig_st_foreach(t, i_renumber_name, (HashDataType )map);
2025/05/31 00:59:04 WARN   cargo:warning=      |                        ^~~~~~~~~~~~~~~
2025/05/31 00:59:04 WARN   cargo:warning=      |                        |
2025/05/31 00:59:04 WARN   cargo:warning=      |                        int (*)(OnigUChar *, NameEntry *, GroupNumMap *) {aka int (*)(unsigned char *, NameEntry *, GroupNumMap *)}
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, NameEntry *, GroupNumMap *)' {aka 'int (*)(unsigned char *, NameEntry *, GroupNumMap *)'}
2025/05/31 00:59:04 WARN   cargo:warning=   55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
2025/05/31 00:59:04 WARN   cargo:warning=      |                               ^~~~~~~~~~~~~~~~
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/st.h:35:18: note: in definition of macro '_'
2025/05/31 00:59:04 WARN   cargo:warning=   35 | # define _(args) args
2025/05/31 00:59:04 WARN   cargo:warning=      |                  ^~~~
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/regparse.c:879:1: note: 'i_renumber_name' declared here
2025/05/31 00:59:04 WARN   cargo:warning=  879 | i_renumber_name(UChar* key ARG_UNUSED, NameEntry* e, GroupNumMap* map)
2025/05/31 00:59:04 WARN   cargo:warning=      | ^~~~~~~~~~~~~~~
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/regparse.c: In function 'callout_name_table_clear':
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/regparse.c:1386:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
2025/05/31 00:59:04 WARN   cargo:warning= 1386 |     onig_st_foreach(t, i_free_callout_name_entry, 0);
2025/05/31 00:59:04 WARN   cargo:warning=      |                        ^~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:59:04 WARN   cargo:warning=      |                        |
2025/05/31 00:59:04 WARN   cargo:warning=      |                        int (*)(st_callout_name_key *, CalloutNameEntry *, void *)
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(st_callout_name_key *, CalloutNameEntry *, void *)'
2025/05/31 00:59:04 WARN   cargo:warning=   55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
2025/05/31 00:59:04 WARN   cargo:warning=      |                               ^~~~~~~~~~~~~~~~
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/st.h:35:18: note: in definition of macro '_'
2025/05/31 00:59:04 WARN   cargo:warning=   35 | # define _(args) args
2025/05/31 00:59:04 WARN   cargo:warning=      |                  ^~~~
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/regparse.c:1370:1: note: 'i_free_callout_name_entry' declared here
2025/05/31 00:59:04 WARN   cargo:warning= 1370 | i_free_callout_name_entry(st_callout_name_key* key, CalloutNameEntry* e,
2025/05/31 00:59:04 WARN   cargo:warning=      | ^~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/regparse.c: In function 'setup_ext_callout_list_values':
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/regparse.c:1884:56: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
2025/05/31 00:59:04 WARN   cargo:warning= 1884 |     onig_st_foreach((CalloutTagTable *)ext->tag_table, i_callout_callout_list_set,
2025/05/31 00:59:04 WARN   cargo:warning=      |                                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:59:04 WARN   cargo:warning=      |                                                        |
2025/05/31 00:59:04 WARN   cargo:warning=      |                                                        int (*)(OnigUChar *, CalloutTagVal,  void *) {aka int (*)(unsigned char *, long int,  void *)}
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, CalloutTagVal,  void *)' {aka 'int (*)(unsigned char *, long int,  void *)'}
2025/05/31 00:59:04 WARN   cargo:warning=   55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
2025/05/31 00:59:04 WARN   cargo:warning=      |                               ^~~~~~~~~~~~~~~~
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/st.h:35:18: note: in definition of macro '_'
2025/05/31 00:59:04 WARN   cargo:warning=   35 | # define _(args) args
2025/05/31 00:59:04 WARN   cargo:warning=      |                  ^~~~
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/regparse.c:1866:1: note: 'i_callout_callout_list_set' declared here
2025/05/31 00:59:04 WARN   cargo:warning= 1866 | i_callout_callout_list_set(UChar* key, CalloutTagVal e, void* arg)
2025/05/31 00:59:04 WARN   cargo:warning=      | ^~~~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/regparse.c: In function 'callout_tag_table_clear':
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/regparse.c:1932:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
2025/05/31 00:59:04 WARN   cargo:warning= 1932 |     onig_st_foreach(t, i_free_callout_tag_entry, 0);
2025/05/31 00:59:04 WARN   cargo:warning=      |                        ^~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:59:04 WARN   cargo:warning=      |                        |
2025/05/31 00:59:04 WARN   cargo:warning=      |                        int (*)(OnigUChar *, CalloutTagVal,  void *) {aka int (*)(unsigned char *, long int,  void *)}
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, CalloutTagVal,  void *)' {aka 'int (*)(unsigned char *, long int,  void *)'}
2025/05/31 00:59:04 WARN   cargo:warning=   55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
2025/05/31 00:59:04 WARN   cargo:warning=      |                               ^~~~~~~~~~~~~~~~
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/st.h:35:18: note: in definition of macro '_'
2025/05/31 00:59:04 WARN   cargo:warning=   35 | # define _(args) args
2025/05/31 00:59:04 WARN   cargo:warning=      |                  ^~~~
2025/05/31 00:59:04 WARN   cargo:warning=oniguruma/src/regparse.c:1922:1: note: 'i_free_callout_tag_entry' declared here
2025/05/31 00:59:04 WARN   cargo:warning= 1922 | i_free_callout_tag_entry(UChar* key, CalloutTagVal e, void* arg ARG_UNUSED)
2025/05/31 00:59:04 WARN   cargo:warning=      | ^~~~~~~~~~~~~~~~~~~~~~~~
2025/05/31 00:59:04 WARN
2025/05/31 00:59:04 WARN   --- stderr
2025/05/31 00:59:04 WARN
2025/05/31 00:59:04 WARN
2025/05/31 00:59:04 WARN   error occurred in cc-rs: Command LC_ALL="C" "cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "/home/build/target/release/build/onig_sys-c6aa338633fc228a/out" "-I" "oniguruma/src" "-DHAVE_UNISTD_H=1" "-DHAVE_SYS_TYPES_H=1" "-DHAVE_SYS_TIME_H=1" "-o" "/home/build/target/release/build/onig_sys-c6aa338633fc228a/out/a445302c6d3dcb51-regparse.o" "-c" "oniguruma/src/regparse.c" with args cc did not execute successfully (status code exit status: 1).
2025/05/31 00:59:04 WARN
2025/05/31 00:59:04 WARN
2025/05/31 00:59:19 ERRO failed to build package: unable to run package xh pipeline: unable to run pipeline: exit status 101
make[1]: *** [Makefile:157: packages/x86_64/xh-0.24.1-r2.apk] Error 1
make[1]: Leaving directory '/home/sergio_durigan_chainguard_dev/os'
make: *** [Makefile:151: package/xh] Error 2
```

--- a/xh.yaml
+++ b/xh.yaml
@@ -1,7 +1,7 @@
 package:
   name: xh
   version: "0.24.1"
-  epoch: 2
+  epoch: 3
   description: Friendly and fast tool for sending HTTP requests.
   copyright:
     - license: MIT
@@ -13,7 +13,6 @@ environment:
       - busybox
       - ca-certificates-bundle
       - cargo-auditable
-      - gcc-14-default
       - rust
 
 pipeline:

--- a/xh/cargobump-deps.yaml
+++ b/xh/cargobump-deps.yaml
@@ -9,3 +9,5 @@ packages:
       version: 0.17.13
     - name: tokio
       version: 1.43.1
+    - name: onig
+      version: 6.5.1

